game-engine/src/test/scala/ConcurrentLogic_Tests.scala

```diff
@@ -0,0 +1,325 @@
+import org.scalatest.FunSuite
+import test.TestHelper
+import za.co.entelect.challenge.game.contracts.Config.Config
+import za.co.entelect.challenge.game.contracts.command.RawCommand
+import za.co.entelect.challenge.game.contracts.commands.CommandFactory
+import za.co.entelect.challenge.game.contracts.game.{CarGamePlayer}
+import za.co.entelect.challenge.game.contracts.map.CarGameMap
+
+class ConcurrentLogic_Tests extends FunSuite{
+  private var nothingCommandText: String = null;
+  private var turnLeftCommandText: String = null;
+  private var turnRightCommandText: String = null;
+
+  private var commandFactory: CommandFactory = null
+
+  private var nothingCommand: RawCommand = null
+  private var turnLeftCommand: RawCommand = null
+  private var turnRightCommand: RawCommand = null
+
+
+  def initialise() = {
+    Config.loadDefault();
+
+    nothingCommandText = Config.NOTHING_COMMAND
+    turnLeftCommandText = Config.TURN_LEFT_COMMAND
+    turnRightCommandText = Config.TURN_RIGHT_COMMAND
+
+    commandFactory = new CommandFactory
+    nothingCommand = commandFactory.makeCommand(nothingCommandText)
+    nothingCommand.setCommand(nothingCommandText)
+    turnLeftCommand = commandFactory.makeCommand(turnLeftCommandText)
+    turnLeftCommand.setCommand(turnLeftCommandText)
+    turnRightCommand = commandFactory.makeCommand(turnRightCommandText)
+    turnRightCommand.setCommand(turnRightCommandText)
+  }
+
+  test("Given players during a race when player 1 turns into player 2 from the left then player 1 position is correctly updated to resolve the collision")
+  {
+    //Arrange: setup players for a collision
+    initialise()
+    val gameMap = TestHelper.initialiseGameWithNoMapObjects()
+    val carGameMap = gameMap.asInstanceOf[CarGameMap]
+
+    val testGamePlayer1 = TestHelper.getTestGamePlayer1()
+    val testCarGamePlayer1 = testGamePlayer1.asInstanceOf[CarGamePlayer]
+    val testGamePlayer1Id = testCarGamePlayer1.getGamePlayerId()
+    testCarGamePlayer1.speed = Config.SPEED_STATE_2 //should be 6
+    val newLaneMidRacePlayer1 = 2
+    val newBlockNumberMidRacePlayer1 = 35
+    TestHelper.putPlayerSomewhereOnTheTrack(carGameMap, testGamePlayer1Id, newLaneMidRacePlayer1, newBlockNumberMidRacePlayer1)
```

Comment:
All of the above could've been moved into setup methods, IMO.

Reply:
Will give the tests another look to remove duplication 👍